### PR TITLE
Ensures that one and only one README file is uploaded and that there is one or more files uploaded for any given Work

### LINKED ISSUE #1772: Validation for Awaiting Approval (end of wizard) Need to check for Readme and Other files explicitly
The user should not be able to mark a submission as complete unless a README file exists AND some other file.

Completion Criteria:
- [x] When a user Click Grant License and Complete the controller should validate the uploaded files
  - [x] There Must be one and only one README.
  - [x] There Must be one or more other files if they are uploading locally
- [x] If the right files do not exists the user will see an error on the page and the work will not update state
  - [x] Readme error: "You Must include a README. [Please upload  one][link to readme step]"
  - [x] File error "You Must include at least one file. [Please upload  one][link to upload step]"

### PR DESCRIPTION
**Pull request by jrgriffiniii**
refs #1772

### CODE REVIEW

**Comment by jrgriffiniii** [comment]
I suspect that the error messages do not directly match what is required, I am very sorry. Once I have these matching, I can open this and request review.

File: app/controllers/works_wizard_controller.rb [hunk 1]
```diff
@@ -158,6 +167,19 @@ def file_location_url
   end
   helper_method :file_location_url
 
+  def rescue_aasm_error
```
Comment (carolyncole) [inline app/controllers/works_wizard_controller.rb:170]: @jrgriffiniii why was this redefined here?

File: app/services/work_validator.rb [hunk 2]
```diff
@@ -161,6 +162,7 @@ def validate_metadata
     def validate_files
       return if @work.resource.migrated
       readme = Readme.new(work, nil)
-      errors.add(:base, "Must provide a README") if readme.blank?
+      errors.add(:base, "You must include a README. <a href='#{work_readme_select_path(work)}'>Please upload one</a>") if readme.blank?
+      errors.add(:base, "You must provide only one README file upload") if work.readme_uploads.length > 1
```
Comment (carolyncole) [inline app/services/work_validator.rb:166]: I believe we will allow the user to add other version of the PDF in files

**Review by carolyncole — CHANGES_REQUESTED** [review]
@jrgriffiniii this PR looks good, but I believe the user can add additional readme formats in the files.  That means the check in the validation should be removed.  As long as there is one the work is valid.  If there are 3 the work should still be valid.

**Review by carolyncole — APPROVED** [review]

**Comment by carolyncole** [comment]
@jrgriffiniii I'm happy to merge this PR.  I want to be certain you are aware that I can continue and submit even if I do not submit any other file than a readme.

**Pull request merged**